### DISCUSSION TITLE
How to log whole request which is going to be sent to openAI?

### DISCUSSION THREAD

**Discussion by WebsheetPlugin** [microsoft/autogen#2331] [Q&A] [answered]
I have curently some weird 500 error responses from OpenAI. Therefore, I wanted to see what exactly Autogen sends to OpenAI in the request. 

Would anyone be able to help me with some instructions on how to do this? 

I use:

self.user_proxy.initiate_chat(
    self.manager, message=body_plain
)

**Comment by jackgerrits** [comment, marked as answer]
Check this out for runtime logging: https://microsoft.github.io/autogen/docs/notebooks/agentchat_logging

**Reply by WebsheetPlugin** [reply]
Thanks a lot! This was what I was looking for.
